mappings/net/minecraft/world/chunk/WorldChunk.mapping

```diff
@@ -59,6 +60,8 @@ CLASS bxn net/minecraft/world/chunk/WorldChunk
 		ARG 4 clearOld
 	METHOD b remove (Lail;)V
 	METHOD c setLoadedToWorld (Z)V
+	METHOD d setUnsaved (Z)V
+		ARG 1 saved
```

Comment:
Either the method should be setSaved or the parameter should be unsaved.

Reply:
The field is set to `true` when a new Entity is added (see WorldChunk.addEntity).

`ChunkSerializer.serialize` calls the method with a value of `false`

   worldChunk14.setUnsaved(false);

`ChunkSerializer.addEntities` calls `addEntity` and then calls the method with the value `true`.

(there is an Enigma bug here where `chunk` has been mistakenly labelled `entity`. I've corrected the name below to properly account for the lambda.)

    private static void writeEntities(CompoundTag tag, WorldChunk chunk) {
        ListTag listTag3 = tag.getList("Entities", 10);
        World world4 = chunk.getWorld();
        for (int var5 = 0; var5 < listTag3.size(); ++var5) {
            CompoundTag compoundTag6 = listTag3.getCompoundTag(var5);
            EntityType.loadEntityWithPassengers(compoundTag6, world4, entity -> {
                chunk.addEntity(entity);
                return entity;
            });
            chunk.setUnsaved(true);
        }

 It is also used in `needsSaving` when checking the `lastSaveTime` hence it is `unsaved` or `dirty` as a certain voldemort might call it.

    @Override
    public boolean needsSaving() {
        return this.shouldSave || (this.unsaved && this.world.getTime() != this.lastSaveTime);
    }

Reply:
I'm just talking about the name of the parameter not matching the method name. Anyone seeing the code `setUnsaved(true)` will think it's marking the chunk as *unsaved*. But according to your parameter name, it's actually marking it as *saved*.

If the parameter is named correctly, the method should be `setSaved`. Otherwise, the parameter should be `unsaved`

Reply:
Oh, derp. Thanks for that.

Reply:
In cases like this I normally just name the parameter `value`, though. I can do the `unsaved` change for now, but maybe later we need to get a standard on what to call parameters in this situation.

Reply:
The standard is to name setter parameters the same as the property they're setting. There are over 1000 setters in yarn all using this convention, it's the convention Mojang uses in their public libraries, and it's the convention used in the JDK.

Reply:
It is the standard? TIL.

~~Well, it makes sense, but I've always used ``value``~~